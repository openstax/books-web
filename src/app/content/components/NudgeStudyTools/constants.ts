export const arrowDesktopHeight = 13;
export const arrowMobileHeight = 6.2;
export const arrowLeftMargin = 1;
export const arrowTopMargin = 1;

export const closeButtonDistanceFromContent = 4;
export const closeButtonMobileMargin = 3;

export const contentMarginTop = 1;

export const spotlightPadding = 1;

export const cookieNudge = {
  // How many times nuge was shown
  counter: 'nudge_study_guides_counter',
  // When nudge was shown
  date: 'nudge_study_guides_date',
  // How many pages user has opened (opening the same page twice counts it as 2)
  pageCounter: 'nudge_study_guides_page_counter',
};

// Show nudge only for users which opened at least 2 pages
export const nudgeStudyToolsMinPageLimit = 2;

// Show nudge every 4 weeks: 4 * 7 days * 24 hours * 60 minutes * 60 seconds * 1000 ms
export const timeIntervalBetweenShowingNudgeInMs = 4 * 7 * 24 * 60 * 60 * 1000;
// Show nudge max 3 times
export const nudgeStudyToolsShowLimit = 3;
// Id of the element which should be highlighted
export const nudgeStudyToolsTargetId = 'nudge-study-tools';
