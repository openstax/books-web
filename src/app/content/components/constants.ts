import theme from '../../theme';

export const bookBannerDesktopBigHeight = 13;
export const bookBannerDesktopMiniHeight = 7;
export const bookBannerMobileBigHeight = 10.4;
export const bookBannerMobileMiniHeight = 6;

export const sidebarDesktopWidth = 33.5;
export const sidebarMobileWidth = 28.8;
export const sidebarTransitionTime = 300;

export const toolbarIconColor = theme.color.primary.gray;
export const toolbarDesktopHeight = 5;
export const toolbarMobileHeight = 4;

export const contentTextWidth = 84;

export const mainContentBackground = '#fff';

export const maxContentGutter = 6;
export const contentWrapperMaxWidth = contentTextWidth + sidebarDesktopWidth + maxContentGutter * 2;
