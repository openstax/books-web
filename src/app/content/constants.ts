import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import theme from '../theme';

export const studyGuidesFeatureFlag = 'studyGuidesEnabled';
export const practiceQuestionsFeatureFlag = 'practiceQuestionsEnabled';

export const maxHighlightsApiPageSize = 200;
export const maxResourcesPerFetch = 10;

export const summaryPageSize = 20;
export const loadMoreDistanceFromBottom = 50;

export const modalQueryParameterName = 'modal';

export const highlightStyles: Array<{label: HighlightColorEnum, passive: string, focused: string}> = [
  {label: HighlightColorEnum.Yellow, passive: '#ffff8a', focused: '#fed200'},
  {label: HighlightColorEnum.Green, passive: '#def99f', focused: '#92d101'},
  {label: HighlightColorEnum.Blue, passive: '#c8f5ff', focused: '#00c3ed'},
  {label: HighlightColorEnum.Purple, passive: '#cbcfff', focused: '#545ec8'},
  {label: HighlightColorEnum.Pink, passive: '#ffc5e1', focused: '#de017e'},
];

const getPrimaryBaseColors = (primaryColors: {[key: string]: {[key: string]: string}}) => {
  const primaryBases: { [key: string]: string } = {};

  // tslint:disable-next-line: max-line-length
  Object.keys(primaryColors).forEach((color) => {primaryBases[color] = primaryColors[color].base; });
  return primaryBases;
};

export const searchButtonStyles: {[key: string]: string} = {
  gray: '#5D6062',
  transparent: 'transparent',
  ...getPrimaryBaseColors(theme.color.primary) as {},
};
