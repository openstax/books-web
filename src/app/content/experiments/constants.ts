export const experiments: { [key: string]: any } = {
  searchButton: [
    'originalButtonEnabled',
    'bannerColorButtonEnabled',
    'grayButtonEnabled',
  ],
};

export const experimentIds: { [key: string]: any } = {
  OCCkMMCZSwW87szzpniCow: 'searchButton',
};
