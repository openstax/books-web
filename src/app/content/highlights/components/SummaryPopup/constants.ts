export const mobilePaddingSides = 1.6;
export const mobilePaddingTopBottom = 1.4;
export const mobileMarginSides = 0.8;
export const mobileMarginTopBottom = 2;
