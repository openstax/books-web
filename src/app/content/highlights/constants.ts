import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import books from '../../../config.books';

export const maxHighlightsApiPageSize = 200;
export const maxResourcesPerFetch = 10;

export const summaryPageSize = 20;
export const loadMoreDistanceFromBottom = 50;

export const cardWidth = 20;
export const cardPadding = 0.8;
export const cardContentMargin = 3;
export const cardFocusedContentMargin = 1;
export const cardMinWindowMargin = 2;
export const cardMarginBottom = 2;

export const highlightStyles: Array<{label: HighlightColorEnum, passive: string, focused: string}> = [
  {label: HighlightColorEnum.Yellow, passive: '#ffff8a', focused: '#fed200'},
  {label: HighlightColorEnum.Green, passive: '#def99f', focused: '#92d101'},
  {label: HighlightColorEnum.Blue, passive: '#c8f5ff', focused: '#00c3ed'},
  {label: HighlightColorEnum.Purple, passive: '#cbcfff', focused: '#545ec8'},
  {label: HighlightColorEnum.Pink, passive: '#ffc5e1', focused: '#de017e'},
];

export const enabledForBooks = Object.keys(books);

export const highlightBlockPadding = 1;
export const highlightIndicatorSizeForBlock = 1.2;
export const highlightIndicatorSize = 0.9;
