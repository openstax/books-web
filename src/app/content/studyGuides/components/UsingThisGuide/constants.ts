export const cookieUTG = 'user_has_seen_UTG';
