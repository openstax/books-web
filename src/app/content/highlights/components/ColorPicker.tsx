import { HighlightColorEnum } from '@openstax/highlighter/dist/api';
import React from 'react';
import { FormattedMessage } from 'react-intl';
import styled from 'styled-components/macro';
import { match, not } from '../../../fpUtils';
import { cardPadding, highlightStyles } from '../constants';
import ColorIndicator from './ColorIndicator';

interface SingleSelectProps {
  color?: string;
  onRemove: () => void;
  onChange: (color: string) => void;
  multiple: false | undefined;
}

interface MultipleSelectProps {
  selected: HighlightColorEnum[];
  onChange: (selected: string[]) => void;
  multiple: true;
}

type Props = (SingleSelectProps | MultipleSelectProps) & {
  className?: string;
  size?: 'small';
};

interface ColorButtonProps {
  name: string;
  checked: boolean;
  size: Props['size'];
  style: typeof highlightStyles[number];
  onClick: () => void;
  className?: string;
}

// tslint:disable-next-line:variable-name
const ColorButton = styled(({className, size, style, ...props}: ColorButtonProps) =>
  <FormattedMessage id={`i18n:highlighting:colors:${style.label}`}>
    {(msg: Element | string) =>
      <ColorIndicator
        style={style}
        size={size}
        title={msg}
        component={<label />}
        className={className}
      >
        <input type='checkbox' {...props} />
      </ColorIndicator>
    }
  </FormattedMessage>
)`
  cursor: pointer;
  margin: 0 ${cardPadding}rem ${cardPadding}rem 0;

  input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
    height: 0;
    width: 0;
  }
`;

// tslint:disable-next-line:variable-name
const ColorPicker = ({className, ...props}: Props) => {

  return <div className={className}>
    {highlightStyles.map((style) => <ColorButton key={style.label}
      name={style.label}
      checked={props.multiple ? props.selected.includes(style.label) : props.color === style.label}
      style={style}
      onChange={() => props.multiple
        ? props.selected.includes(style.label)
          ? props.onChange(props.selected.filter(not(match(style.label))))
          : props.onChange([...props.selected, style.label])
        : props.color === style.label
          ? props.onRemove()
          : props.onChange(style.label)}
    />)}
  </div>;
};

export default styled(ColorPicker)`
  display: flex;
  flex-direction: row;
`;
