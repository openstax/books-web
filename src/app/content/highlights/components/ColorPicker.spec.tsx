import React from 'react';
import renderer from 'react-test-renderer';
import { highlightStyles } from '../constants';
import ColorPicker from './ColorPicker';

describe('ColorPicker', () => {
  it('matches snapshot no selection', () => {
    const component = renderer.create(<ColorPicker onChange={jest.fn()} onRemove={jest.fn()} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('checks selection', () => {
    const component = renderer.create(
      <ColorPicker color={highlightStyles[0].label} onChange={jest.fn()} onRemove={jest.fn()} />
    );

    const [first, ...rest] = component.root.findAllByType('input');
    expect(first.props.checked).toEqual(true);
    rest.forEach((input) => expect(input.props.checked).toEqual(false));
  });

  it('checks selection (multiple)', () => {
    const component = renderer.create(
      <ColorPicker multiple selected={[highlightStyles[0].label]} onChange={jest.fn()} onRemove={jest.fn()} />
    );

    const [first, ...rest] = component.root.findAllByType('input');
    expect(first.props.checked).toEqual(true);
    rest.forEach((input) => expect(input.props.checked).toEqual(false));
  });

  it('calls update when changing selection', () => {
    const onChange = jest.fn();
    const onRemove = jest.fn();

    const component = renderer.create(
      <ColorPicker color={highlightStyles[0].label} onChange={onChange} onRemove={onRemove} />
    );

    const [, second] = component.root.findAllByType('input');

    second.props.onChange();

    expect(onRemove).not.toHaveBeenCalled();
    expect(onChange).toHaveBeenCalledWith(highlightStyles[1].label);
  });

  it('calls update when changing selection (multiple)', () => {
    const onChange = jest.fn();
    const onRemove = jest.fn();

    const component = renderer.create(
      <ColorPicker multiple selected={[highlightStyles[0].label]} onChange={onChange} onRemove={onRemove} />
    );

    const [, second] = component.root.findAllByType('input');

    second.props.onChange();

    expect(onRemove).not.toHaveBeenCalled();
    expect(onChange).toHaveBeenCalledWith([highlightStyles[0].label, highlightStyles[1].label]);
  });

  it('calls remove when changing selection', () => {
    const onChange = jest.fn();
    const onRemove = jest.fn();

    const component = renderer.create(
      <ColorPicker color={highlightStyles[0].label} onChange={onChange} onRemove={onRemove} />
    );

    const [first] = component.root.findAllByType('input');

    first.props.onChange();

    expect(onRemove).toHaveBeenCalled();
    expect(onChange).not.toHaveBeenCalled();
  });

  it('calls remove when changing selection (multiple)', () => {
    const onChange = jest.fn();
    const onRemove = jest.fn();

    const component = renderer.create(
      <ColorPicker multiple selected={[highlightStyles[0].label]} onChange={onChange} onRemove={onRemove} />
    );

    const [first] = component.root.findAllByType('input');

    first.props.onChange();

    expect(onRemove).not.toHaveBeenCalled();
    expect(onChange).toHaveBeenCalledWith([]);
  });
});
