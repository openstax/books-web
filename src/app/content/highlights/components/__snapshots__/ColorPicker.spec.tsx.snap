// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker matches snapshot no selection 1`] = `
.c5 {
  display: inline-block;
  vertical-align: -.125em;
  overflow: hidden;
}

.c4 {
  display: none;
}

.c2 {
  background-color: #ffff8a;
  border: 1px solid #fed200;
  height: 2.4rem;
  width: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2rem;
}

.c2 .c3 {
  height: 1.6rem;
  width: 1.6rem;
}

.c2 input:checked + .c3 {
  color: #fed200;
  display: block;
}

.c6 {
  background-color: #def99f;
  border: 1px solid #92d101;
  height: 2.4rem;
  width: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2rem;
}

.c6 .c3 {
  height: 1.6rem;
  width: 1.6rem;
}

.c6 input:checked + .c3 {
  color: #92d101;
  display: block;
}

.c7 {
  background-color: #c8f5ff;
  border: 1px solid #00c3ed;
  height: 2.4rem;
  width: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2rem;
}

.c7 .c3 {
  height: 1.6rem;
  width: 1.6rem;
}

.c7 input:checked + .c3 {
  color: #00c3ed;
  display: block;
}

.c8 {
  background-color: #cbcfff;
  border: 1px solid #545ec8;
  height: 2.4rem;
  width: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2rem;
}

.c8 .c3 {
  height: 1.6rem;
  width: 1.6rem;
}

.c8 input:checked + .c3 {
  color: #545ec8;
  display: block;
}

.c9 {
  background-color: #ffc5e1;
  border: 1px solid #de017e;
  height: 2.4rem;
  width: 2.4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 2rem;
}

.c9 .c3 {
  height: 1.6rem;
  width: 1.6rem;
}

.c9 input:checked + .c3 {
  color: #de017e;
  display: block;
}

.c1 {
  cursor: pointer;
  margin: 0 0.8rem 0.8rem 0;
}

.c1 input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.c0 {
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

<div
  className="c0"
  tabIndex={-1}
>
  <label
    className="c1 c2"
    title="yellow"
  >
    <input
      checked={false}
      name="yellow"
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      className="c3 c4 c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </label>
  <label
    className="c1 c6"
    title="green"
  >
    <input
      checked={false}
      name="green"
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      className="c3 c4 c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </label>
  <label
    className="c1 c7"
    title="blue"
  >
    <input
      checked={false}
      name="blue"
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      className="c3 c4 c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </label>
  <label
    className="c1 c8"
    title="purple"
  >
    <input
      checked={false}
      name="purple"
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      className="c3 c4 c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </label>
  <label
    className="c1 c9"
    title="pink"
  >
    <input
      checked={false}
      name="pink"
      onChange={[Function]}
      type="checkbox"
    />
    <svg
      aria-hidden="true"
      className="c3 c4 c5"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 512 512"
    >
      <path
        d="M173.898 439.404l-166.4-166.4c-9.997-9.997-9.997-26.206 0-36.204l36.203-36.204c9.997-9.998 26.207-9.998 36.204 0L192 312.69 432.095 72.596c9.997-9.997 26.207-9.997 36.204 0l36.203 36.204c9.997 9.997 9.997 26.206 0 36.204l-294.4 294.401c-9.998 9.997-26.207 9.997-36.204-.001z"
        fill="currentColor"
      />
    </svg>
  </label>
</div>
`;
